File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.js
/**
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
var checkArithmeticSubarrays = function (nums, l, r) {
    const list = [];
    for (let i = 0; i < l.length; i++) 
        list.push(isArithmaticSeq(nums, l[i], r[i]));
    
    return list;
};

var isArithmaticSeq = function (arr, l, r) {

    const nums = arr.slice(l, r + 1);

    if (nums.length <= 2) return true;

    nums.sort((a, b) => a - b);

    const diff = nums[1] - nums[0];
    for (let i = 2; i < nums.length; i++) 
        if (diff !== nums[i] - nums[i - 1]) 
            return false;
        
    return true;
}

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> boolList = new ArrayList<>();
        for (int i = 0; i < l.length; i++) 
            boolList.add(isArithmeticSeq(nums, l[i], r[i]));

        return boolList;
    }

    private static boolean isArithmeticSeq(int[] arr, int l, int r) {
        int n = r - l + 1;
        if (n <= 2) return true;

        int[] nums = Arrays.copyOfRange(arr, l, r + 1);
        Arrays.sort(nums);

        int diff = nums[1] - nums[0];

        for (int i = 2; i < nums.length; i++) 
            if (nums[i] - nums[i - 1] != diff) 
                return false;

        return true;
    }

}